tests/integration/components/au-button-test.js

```diff
@@ -23,4 +23,29 @@ module('Integration | Component | au-button', function (hooks) {
 
     assert.dom(this.element).hasText('template block text');
   });
+
+  test('it hides the text when @hideText is set to true', async function (assert) {
+    await render(hbs`
+      <AuButton
+        @hideText={{true}}
+      >
+        template block text
+      </AuButton>
+    `);
+
+    assert.dom('.au-u-hidden-visually').exists();
+  });
+
+  test('it adds the icon-only modifier class when an icon is defined and @hideText is set to true', async function (assert) {
+    await render(hbs`
+      <AuButton
+        @icon="plus"
+        @hideText={{true}}
+      >
+        template block text
+      </AuButton>
+    `);
+
+    assert.dom('button').hasClass('au-c-button--icon-only');
```

Comment:
Is using `.dom('button')` good practice by the way or should I use a data-test selector here?

Reply:
I think it's fine for simple tests like this and in places where the element isn't an implementation detail of the component 👍.